helpers/helpers.go

```diff
@@ -164,6 +164,11 @@ func ParseArgs(args []string, argFlags *flag.FlagSet) error {
 	pos, args = getPositional(args)
 	// append positional args back at the end of args
 	args = append(args, pos...)
+	for _, item := range args {
+		if item == "-config" || item == "--config" {
+			return fmt.Errorf("the config flag should come before the subcommand. Eg 'sda-cli -config s3cfg %s'", argFlags.Name())
+		}
+	}
```

Comment:
There are multiple ways of doing this below is another example that uses the slice index.
Both methods are equally valid but here we don't need to iterate through the entire slice of arguments.
```
	if idx := slices.IndexFunc(args, func(arg string) bool { return strings.Contains(arg, "-config") }); idx != 0 {
		return fmt.Errorf("the config flag should come before the subcommand. Eg 'sda-cli -config s3cfg %s'", argFlags.Name())
	}
```

Reply:
Thanks, that's good to know!

I will go with the current version, since it is approved, tested etc. But very nice to know for the future.